src/node_report.cc

```diff
@@ -0,0 +1,790 @@
+  } else
+#endif
+  if (IsWindows8OrGreater()) {
+    fprintf(fp,"8 ");
+  } else if (IsWindows7OrGreater()) {
+    fprintf(fp,"7 ");
+  } else if (IsWindowsXPOrGreater()) {
+    fprintf(fp,"XP ");
+  }
+  if (IsWindowsServer()) {
+    fprintf(fp,"Server\n");
+  } else {
+    fprintf(fp,"Client\n");
+  }
+  TCHAR  infoBuf[256];
+  DWORD  bufCharCount = 256;
+  if (GetComputerName(infoBuf, &bufCharCount)) {
+    fprintf(fp,"Machine name: %s %s\n", infoBuf);
+  }
+#else
+  struct utsname os_info;
+  if (uname(&os_info) == 0) {
+    fprintf(fp,"\nOS version: %s %s %s",os_info.sysname, os_info.release, os_info.version);
+    fprintf(fp,"\nMachine: %s %s\n", os_info.nodename, os_info.machine);
+  }
+#endif
+
+  // Print native process ID
+  fprintf(fp, "\nProcess ID: %d\n", pid);
+  fflush(fp);
+
+// Print summary JavaScript stack trace
+  fprintf(fp, "\n================================================================================");
+  fprintf(fp, "\n==== JavaScript Stack Trace ====================================================\n\n");
+
+#ifdef _WIN32
+  switch (event) {
+  case kFatalError:
+    // Stack trace on fatal error not supported on Windows
+    fprintf(fp, "No stack trace available\n");
+    break;
+  default:
+    // All other events, print the stack using StackTrace::StackTrace() and GetStackSample() APIs
+    PrintStackFromStackTrace(fp, isolate, event);
+    break;
+  }  // end switch(event)
+#else  // Unix, OSX
+  switch (event) {
+  case kException:
+  case kJavaScript:
+    // Print the stack using Message::PrintCurrentStackTrace() API
+    Message::PrintCurrentStackTrace(isolate, fp);
+    break;
+  case kFatalError:
+#if NODE_VERSION_AT_LEAST(6, 0, 0)
+    if (!strncmp(location,"MarkCompactCollector", sizeof("MarkCompactCollector") - 1)) {
+      fprintf(fp, "V8 running in GC - no stack trace available\n");
+    } else {
+      Message::PrintCurrentStackTrace(isolate, fp);
+    }
+#else
+    fprintf(fp, "No stack trace available\n");
+#endif
+    break;
+  case kSignal_JS:
+  case kSignal_UV:
+    // Print the stack using StackTrace::StackTrace() and GetStackSample() APIs
+    PrintStackFromStackTrace(fp, isolate, event);
+    break;
+  }  // end switch(event)
+#endif
+  fflush(fp);
+
+  // Print native stack backtrace
+  PrintNativeBacktrace(fp);
+  fflush(fp);
+
+  // Print V8 Heap and Garbage Collector information
+  PrintGCStatistics(fp, isolate);
+  fflush(fp);
+
+  // Print OS and current thread resource usage
+#ifndef _WIN32
+  PrintResourceUsage(fp);
+  fflush(fp);
+#endif
+
+  // Print libuv handle summary (TODO: investigate failure on Windows)
+#ifndef _WIN32
+  fprintf(fp, "\n================================================================================");
+  fprintf(fp, "\n==== Node.js libuv Handle Summary ==============================================\n");
+  fprintf(fp,"\n(Flags: R=Ref, A=Active, I=Internal)\n");
+  fprintf(fp,"\nFlags Type     Address\n");
+  uv_print_all_handles(NULL, fp);
```

Comment:
Be aware that `uv_print_all_handles()` "...is meant for ad hoc debugging, there is no API/ABI stability guarantees."  (From the documentation.)